### PR TITLE
Update RelationshipBlock.php

### LINKED ISSUE #17: Expired relationships being displayed
The contact summary works well, however I've noticed a bug related to - expired - relationships still appearing. I would like to propose a verification be made that the relationship is current/active before it gets displayed.

If there's an interest to display - expired - relationships, then there should be an option to enable that.

Thanks!

### PR DESCRIPTION
**Pull request by vitius1**
Fixes #17 

When the relationship expires, the "is_active" field not change. So expired relationships are displayed. I add condition for end_date relationship. Now you will see relationships that have no end date or the end date has not yet occurred.

### CODE REVIEW

**Comment by colemanw** [comment]
This seems reasonable to me, although there are a couple whitespace issues in the patch.

What do you think @eileenmcnaughton?

**Comment by eileenmcnaughton** [comment]
@colemanw I don't have a problem with this in principle - this hasn't been a thing for us due to our data set so it won't affect us either way

File: CRM/Relationshipblock/Utils/RelationshipBlock.php [hunk 1]
```diff
@@ -23,11 +23,12 @@ public static function getExistingRelationships($contactID) {
       'contact_id_b' => $contactID,
       'return' => ['id', 'relationship_type_id', 'contact_id_a', 'contact_id_b',
         'contact_id_a.display_name', 'contact_id_b.display_name', 'contact_id_a.is_deceased',
-        'contact_id_b.is_deceased'],
+        'contact_id_b.is_deceased', 'end_date'],
       'options' => ['limit' => 0, 'sort' => 'relationship_type_id.label_a_b ASC', 'or' => [['contact_id_a', 'contact_id_b']]],
     ]);
     $ret = [];
     foreach ($existingRelationships['values'] as $rel) {
+      if ( !isset($rel['end_date']) || ($rel['end_date'] < $now )) {
```
Comment (eileenmcnaughton) [inline CRM/Relationshipblock/Utils/RelationshipBlock.php]: do you need to strtime these for more reliable comparison?

Comment (vitius1) [reply CRM/Relationshipblock/Utils/RelationshipBlock.php]: Yes, I can use cron job "Disable expired relationships" for update "is_active" field for relationships, but this is I think better.

Comment (colemanw) [reply CRM/Relationshipblock/Utils/RelationshipBlock.php]: @vitius1 I spotted a bug on this line. The variable `$now` has not been defined.

Comment (vitius1) [reply CRM/Relationshipblock/Utils/RelationshipBlock.php]: For me it worked. But you can change for "strftime()". Maybe only on my website I have defined $now.

Comment (colemanw) [reply CRM/Relationshipblock/Utils/RelationshipBlock.php]: According to [the docs](http://php.net/manual/en/function.strftime.php), you are using that function incorrectly by calling it with no params, as 1 param is required.

Comment (vitius1) [reply CRM/Relationshipblock/Utils/RelationshipBlock.php]: @colemanw I'm sorry for this. I don't know well this function. But i try it, and it worked somehow. For correct syntax, it need to know the format of date for params. It works without params, but it's not correct.

Comment (vitius1) [reply CRM/Relationshipblock/Utils/RelationshipBlock.php]: @colemanw I find the correct format of date. Here it is: strftime("%y-%m-%d").

Comment (colemanw) [reply CRM/Relationshipblock/Utils/RelationshipBlock.php]: We don't usually use `strftime` in CiviCRM, I think it's more reliable to do 

    if (!isset($rel['end_date']) || (strtotime($rel['end_date']) < time() )) {

Comment (vitius1) [reply CRM/Relationshipblock/Utils/RelationshipBlock.php]: @colemanw That was first time, when I worked with time in PHP and strftime was the first thing i found.

**Comment by mattwire** [comment]
@colemanw Where are we with this - are you happy with it now?

**Comment by colemanw** [comment]
@mattwire I've rebased it into #25

**Pull request closed without merging**